### PR TITLE
Add release tag support

### PR DESCRIPTION
**Pull request by TravisEz13**

### CODE REVIEW

**Comment by TravisEz13** [comment]
@daxian-dbw

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 [hunk 1]
```diff
@@ -1,38 +1,56 @@
...
-    Copy-Item "$($appImage.FullName)" "$destination" -force
-}
+# PowerShell Script to clone, build and package PowerShell from specified form and branch
+# Script is intented to use in Docker containers
+# Ensure PowerShell is available in the provided image
+
+param (
+	[string] $fork = "powershell",
+	[string] $branch = "master",
+	[string] $location = "/powershell",
```
Comment (daxian-dbw) [inline Images/GenericLinuxFiles/PowerShellPackage.ps1]: indentations here need to be fixed.

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 [hunk 2]
```diff
@@ -1,38 +1,56 @@
...
+    [switch]$AppImage
+)
+
+$releaseTagParam = @{}
+if($ReleaseTag)
+{
+    $releaseTagParam = @{ 'ReleaseTag' = $ReleaseTag }
+}
+
+git clone --quiet https://github.com/$fork/powershell.git -b $branch
```
Comment (daxian-dbw) [inline Images/GenericLinuxFiles/PowerShellPackage.ps1]: The target location is not specified for `git clone`.

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 [hunk 3]
```diff
@@ -1,38 +1,56 @@
...
+Push-Location
+Set-Location "$location"
+git submodule update --init --recursive --quiet
+Import-Module "$location/build.psm1"
+Start-PSBootstrap -Package -NoSudo
+$output = Split-Path -Parent (Get-PSOutput -Options (New-PSOptions -Publish))
```
Comment (daxian-dbw) [inline Images/GenericLinuxFiles/PowerShellPackage.ps1]: > New-PSOptions -Publish

`-Publish` is unnecessary now, better not use it.

Comment (daxian-dbw) [reply Images/GenericLinuxFiles/PowerShellPackage.ps1]: Minor comment: I think we probably should replace `(Get-PSOutput -Options (New-PSOptions -Publish))` with `(New-PSOptions).Output`. It slightly improves readability :)

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 [hunk 4]
```diff
@@ -1,38 +1,56 @@
...
+Start-PSBuild -Crossgen -PSModuleRestore @releaseTagParam
+
+Start-PSPackage @releaseTagParam
+if($AppImage.IsPresent)
+{
+    Start-PSPackage -Type AppImage @releaseTagParam
+}
+
+Pop-Location
```
Comment (daxian-dbw) [inline Images/GenericLinuxFiles/PowerShellPackage.ps1]: Minor comment: maybe `pop-location` should be put in `finally` block?

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 [hunk 5]
```diff
@@ -1,38 +1,56 @@
...
+
+$linuxPackages = Get-ChildItem "$location/powershell*" -Include *.deb,*.rpm
+    
+foreach($linuxPackage in $linuxPackages) 
+{ 
+    Copy-Item "$($linuxPackage.FullName)" "$destination" -force
```
Comment (daxian-dbw) [inline Images/GenericLinuxFiles/PowerShellPackage.ps1]: How about just `copy-item $linuxPackage.FullName $destination -force`?

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 [hunk 6]
```diff
@@ -1,38 +1,56 @@
...
+
+$releaseTagParam = @{}
+if($ReleaseTag)
+{
+    $releaseTagParam = @{ 'ReleaseTag' = $ReleaseTag }
+}
+
+git clone --quiet https://github.com/$fork/powershell.git -b $branch
+Push-Location
+Set-Location "$location"
```
Comment (daxian-dbw) [inline Images/GenericLinuxFiles/PowerShellPackage.ps1]: `"$location"` can be replaced by `$location`

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 [hunk 7]
```diff
@@ -1,38 +1,56 @@
...
+$linuxPackages = Get-ChildItem "$location/powershell*" -Include *.deb,*.rpm
+    
+foreach($linuxPackage in $linuxPackages) 
+{ 
+    Copy-Item "$($linuxPackage.FullName)" "$destination" -force
+}
+
+if($AppImage.IsPresent)
+{
+    $appImages = Get-ChildItem -Path "$location" -Filter "*.AppImage"
```
Comment (daxian-dbw) [inline Images/GenericLinuxFiles/PowerShellPackage.ps1]: same here, no need to put `$location` in double quotes.

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 [hunk 8]
```diff
@@ -1,38 +1,56 @@
...
+    foreach($appImage in $appImages) 
+    { 
+        Copy-Item "$($appImage.FullName)" "$destination" -force
```
Comment (daxian-dbw) [inline Images/GenericLinuxFiles/PowerShellPackage.ps1]: Same here, just `Copy-Item $appImage.FullName $destination -force`

File: psrelease.psm1 [hunk 9]
```diff
@@ -58,6 +58,23 @@ function Get-Destination
...
 
+function Get-TempFolder
+{
+    $tempPath = $env:TEMP
+    if($env:AGENT_TEMPDIRECTORY)
+    {
+        $tempPath = $env:AGENT_TEMPDIRECTORY
+    }
+
+    $tempFolder = Join-Path -Path $tempPath -ChildPath ([System.IO.Path]::GetRandomFileName())
```
Comment (daxian-dbw) [inline psrelease.psm1:70]: Why not use `[System.IO.Path]::GetTempPath()` for `$tempPath`? Is there a particular reason to use `$env:AGENT_TEMPDIRECTORY`?

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 (hunk 1 above)
Comment (TravisEz13) [reply Images/GenericLinuxFiles/PowerShellPackage.ps1]: #resolved

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 (hunk 2 above)
Comment (TravisEz13) [reply Images/GenericLinuxFiles/PowerShellPackage.ps1]: #resolved

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 (hunk 6 above)
Comment (TravisEz13) [reply Images/GenericLinuxFiles/PowerShellPackage.ps1]: #resolved

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 (hunk 3 above)
Comment (TravisEz13) [reply Images/GenericLinuxFiles/PowerShellPackage.ps1]: Not used, removed

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 (hunk 4 above)
Comment (TravisEz13) [reply Images/GenericLinuxFiles/PowerShellPackage.ps1]: #resolved

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 (hunk 7 above)
Comment (TravisEz13) [reply Images/GenericLinuxFiles/PowerShellPackage.ps1]: #resolved

File: Images/GenericLinuxFiles/PowerShellPackage.ps1 (hunk 8 above)
Comment (TravisEz13) [reply Images/GenericLinuxFiles/PowerShellPackage.ps1]: #resolved

File: psrelease.psm1 (hunk 9 above)
Comment (TravisEz13) [reply psrelease.psm1:70]: You are not supposed to use temp on VSTS machines as they are not reset to a base image between builds.  Only `env:AGENT_TEMPDIRECTORY` is cleaned.  I'm falling back to `$env:TEMP` when it doesn't exist for local testing (changed to `[System.IO.Path]::GetTempPath()`)

Added a comment to clarify why.

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**